app/src/test/java/com/duckduckgo/app/pixels/EnqueuedPixelWorkerTest.kt

```diff
@@ -101,6 +107,30 @@ class EnqueuedPixelWorkerTest {
         enqueuedPixelWorker.onStateChanged(lifecycleOwner, Lifecycle.Event.ON_START)
         enqueuedPixelWorker.onStateChanged(lifecycleOwner, Lifecycle.Event.ON_START)
 
+        verify(pixel).fire(
+            AppPixelName.APP_LAUNCH_LEGACY,
+            mapOf(Pixel.PixelParameter.WEBVIEW_VERSION to "91")
+        )
+
+        verify(pixel, times(0)).fire(
+            AppPixelName.APP_LAUNCH,
+            mapOf(Pixel.PixelParameter.WEBVIEW_VERSION to "91")
+        )
+    }
+
+    @Test
+    fun whenOnStartAndLaunchByFireActionFollowedByAppLaunchAfterGracePeriodThenSendOneAppLaunchPixel() {
+        whenever(unsentForgetAllPixelStore.pendingPixelCountClearData).thenReturn(1)
+        whenever(unsentForgetAllPixelStore.lastClearTimestamp).thenReturn(System.currentTimeMillis())
+        whenever(webViewVersionProvider.getMajorVersion()).thenReturn("91")
+
+        enqueuedPixelWorker.onStateChanged(lifecycleOwner, Lifecycle.Event.ON_CREATE)
+        enqueuedPixelWorker.onStateChanged(lifecycleOwner, Lifecycle.Event.ON_START)
+
+        Thread.sleep(10500)
```

Comment:
10s is a long time, could we not add the grace period as a parameter in the constructor? Something like this:

```
class EnqueuedPixelWorker @Inject constructor(
    private val workManager: WorkManager,
    private val pixel: Provider<Pixel>,
    private val unsentForgetAllPixelStore: UnsentForgetAllPixelStore,
    private val webViewVersionProvider: WebViewVersionProvider,
    private val APP_RESTART_CAUSED_BY_FIRE_GRACE_PERIOD: Long = 10_000L
)
```

Then we can pass it a much shorter value for the test.

Reply:
I like that @joshliebe, nice suggestion,

Reply:
I don't like it 🤣 
Two issues with that constant in the constructor. It's a workaround in production code to a bad test. But more importantly, it will force us to provide a `Long` value into DI as well, which also makes kitties sad.

The class already has the `UnsentForgetAllPixelStore`. That class can be easily faked so that we don't need a `sleep` in the test